cosmo_tester/framework/util.py

```diff
@@ -414,44 +402,113 @@ def generate_ca_cert(ca_cert_path, ca_key_path):
     ])
 
 
-class ExecutionWaiting(Exception):
-    """
-    raised by `wait_for_execution` if it should be retried
-    """
+class ExecutionTimeout(Exception):
+    """Raised by `wait_for_execution` if the execution takes too long."""
 
 
 class ExecutionFailed(Exception):
-    """
-    raised by `wait_for_execution` if a bad state is reached
-    """
-
-
-def retry_if_not_failed(exception):
-    return not isinstance(exception, ExecutionFailed)
+    """Raised by `wait_for_execution` if the execution fails."""
 
 
-@retrying.retry(
-    stop_max_delay=5 * 60 * 1000,
-    wait_fixed=10000,
-    retry_on_exception=retry_if_not_failed,
-)
-def wait_for_execution(manager, execution, logger):
+def wait_for_execution(manager, execution, logger, tenant=None,
+                       new_password=None, timeout=(5*60)):
```

Comment:
the `timeout=(5*60)` is odd (parens?!) - maybe we can run CI's flake8 in a stricter mode? (this is E226, but that is ignored by flake8 by default

Reply:
Updating to remove brackets. I think it was bracketed wherever I was refactoring it from, so I left it like that but yeah, not required here.